### ISSUE TITLE
v5 tokenizers of segment-embedding models (ALBERT, XLNet, Funnel, Splinter, RemBERT, RoFormer, HerBERT) no longer return token_type_ids by default

### ISSUE THREAD

**Issue by 2sumtech** [huggingface/transformers#48128]
### System Info

transformers 5.16.0.dev0 (main @ decba1d), Python 3.11.14, macOS arm64, torch 2.13.0 (CPU-only)

### Who can help?

@ArthurZucker @itazap

### Reproduction

```python
from transformers import AlbertTokenizer, BertTokenizer
albert, bert = AlbertTokenizer(), BertTokenizer()          # same via from_pretrained
print(sorted(albert("hello", "world").keys()))             # ['attention_mask', 'input_ids']  <- missing
print(sorted(bert("hello", "world").keys()))               # ['attention_mask', 'input_ids', 'token_type_ids']
print(albert("hello", "world", return_token_type_ids=True)["token_type_ids"])  # [0, 0, 0, 1, 1] <- capable, just dropped
```

In v4.57.0, `PreTrainedTokenizerBase.model_input_names` defaulted to `["input_ids", "token_type_ids", "attention_mask"]`; on current main it defaults to `["input_ids", "attention_mask"]` and per-model lists were re-declared. Seven tokenizers whose models consume segment embeddings were left without `"token_type_ids"`: **ALBERT, Funnel, HerBERT, RemBERT, RoFormer, Splinter, XLNet**. Since output keys are gated by `"token_type_ids" in self.model_input_names` (`src/transformers/tokenization_utils_tokenizers.py:758-759`), encoding a sentence pair silently omits `token_type_ids`; the models fall back to all-zero segment ids — silent accuracy degradation on pair classification / QA, no error. Reproduced end-to-end via `from_pretrained`/`AutoTokenizer` with a hub-style `tokenizer_config.json` (which never contains `model_input_names`), and `AlbertModel` outputs genuinely change when segment ids are supplied.

BERT, DeBERTa, TAPAS, LayoutLM kept the triple in v5; DistilBERT deliberately opts out (correct); Funnel's v5 tokenizer even preserves its special `cls:2` segment id in its `TemplateProcessing` — currently unreachable by default. The change is not mentioned in MIGRATION_GUIDE_V5.md, which suggests a propagation miss rather than intent.

### Expected behavior

Sentence-pair encodes on these seven tokenizers return `token_type_ids` by default, as in v4 and as `BertTokenizer` still does.

I have a fix ready (restore the per-model `model_input_names` triple, pattern of bert/deberta/tapas, + offline regression test) and will open a PR.
